cg_lims/status_db_api.py

```diff
@@ -1,20 +1,34 @@
-import json
-
 import requests
+from typing import List
+from urllib.parse import urljoin
 
 from cg_lims.exceptions import LimsError
+from cg_lims.models.sequencing_metrics import SequencingMetrics
 
 
-class StatusDBAPI(object):
-    def __init__(self, url=None):
-        self.url = url
+class StatusDBAPI:
+    def __init__(self, base_url):
+        self.base_url = base_url
 
-    def apptag(self, tag_name, key=None, entry_point="/applications"):
+    def _get(self, endpoint, key=None):
+        url = urljoin(self.base_url, endpoint)
         try:
-            res = requests.get(self.url + entry_point + "/" + tag_name)
+            response = requests.get(url)
+            response.raise_for_status()
+            data = response.json()
             if key:
-                return json.loads(res.text)[key]
-            else:
-                return json.loads(res.text)
-        except ConnectionError:
-            raise LimsError(message="No connection to clinical-api!")
+                return data.get(key)
+            return data
+        except requests.RequestException as e:
+            raise LimsError(f"Failed to get data from {url}, {e}")
+
+    def apptag(self, tag_name, key=None):
+        app_tag_endpoint: str = f"/applications/{tag_name}"
+        return self._get(app_tag_endpoint, key)
+
+    def get_sequencing_metrics_for_flow_cell(
+        self, flow_cell_name: str
+    ) -> List[SequencingMetrics]:
+        metrics_endpoint: str = f"/flowcells/{flow_cell_name}/sequencing_metrics"
+        metrics_data = self._get(metrics_endpoint)
```

Comment:
```suggestion
        metrics_json = self._get(metrics_endpoint)
```
We get a json back, maybe this makes it clearer what we are working with.

Reply:
And/or a type hint

Reply:
The return type of `_get` is `Any` (the same as `response.json()´), so not adding it here

Reply:
Actually, lets add the expected types since the actual type is `Any`